src/event_handlers.rs

```diff
@@ -542,13 +545,44 @@ pub async fn handle_floodsub_event(
                 });
             } else if let Ok(relay_msg) = serde_json::from_slice::<crate::types::RelayMessage>(&msg.data) {
                 debug!("Received relay message from {}: {}", msg.source, relay_msg.message_id);
-                // For now, just log the relay message - full implementation will come in next phase
-                ui_logger.log(format!(
-                    "📡 Relay message received from {} (ID: {}, hops: {})",
-                    msg.source, 
-                    &relay_msg.message_id[..8], // Show first 8 chars of ID
-                    relay_msg.hop_count
-                ));
+                
+                // Process relay message if relay service is available
+                if let Some(relay_svc) = relay_service {
+                    match relay_svc.process_relay_message(&relay_msg) {
+                        Ok(crate::relay::RelayAction::DeliverLocally(direct_msg)) => {
+                            ui_logger.log(format!(
+                                "💬 Relay message delivered: {} -> {}: {}",
+                                direct_msg.from_name,
+                                direct_msg.to_name, 
+                                direct_msg.message
+                            ));
+                            debug!("Relay message decrypted and delivered locally: {}", relay_msg.message_id);
+                        }
+                        Ok(crate::relay::RelayAction::ForwardMessage(forward_msg)) => {
+                            // TODO: Re-broadcast the forwarded message via floodsub
```

Comment:
Is this TODO something that needs to be done to make the relay work? If so please implement so this ticket is complete.  If it is already implemented then please remove the comment